GATEWAY/src/providers/tilesApi.service.ts

```diff
@@ -8,6 +8,8 @@ import { Application, LoginData, VirtualTile } from './utils.service';
 
 @Injectable()
 export class TilesApi {
+  public flagThen: boolean = false;
```

Comment:
Because I am not able to test the code without these flags
I can explain this in detail later, but for now it is vital for the testing